src/main/java/ru/yandex/practicum/filmorate/controllers/UserController.java

```diff
@@ -21,28 +20,27 @@ public class UserController {
     public UserController(UserService userService){
         this.userService = userService;
     }
-
-    @GetMapping
-    public List<User> getUsers(){
-        return userService.getUsers();
+    @GetMapping("/{id}")
+    public User getUser(@PathVariable("id") Integer userId){
+        return userService.getUserById(userId);
     }
-
     @PostMapping
     public User addUser(@Valid @RequestBody User user){
```

Comment:
Если передать неверный запрос, то падает на уровне storage. Нужно сделать проверку входных данных на уровне контроллера или сервиса, т.к. обращение к базе данных это всегда медленная операция.

java.lang.NullPointerException: Cannot invoke "java.time.LocalDate.getYear()" because "date" is null
	at java.sql/java.sql.Date.valueOf(Date.java:291) ~[java.sql:na]
	at ru.yandex.practicum.filmorate.storage.daoImpl.DaoUserStorage.lambda$addUser$0(DaoUserStorage.java:48) ~[classes/:na]
	at org.springframework.jdbc.core.JdbcTemplate.execute(JdbcTemplate.java:649) ~[spring-jdbc-5.3.23.jar:5.3.23]
	at org.springframework.jdbc.core.JdbcTemplate.update(JdbcTemplate.java:991) ~[spring-jdbc-5.3.23.jar:5.3.23]
	at ru.yandex.practicum.filmorate.storage.daoImpl.DaoUserStorage.addUser(DaoUserStorage.java:43) ~[classes/:na]
	at ru.yandex.practicum.filmorate.services.UserService.addUser(UserService.java:28) ~[classes/:na]
	at ru.yandex.practicum.filmorate.controllers.UserController.addUser(UserController.java:29)

Reply:
Добавил атрибуты в полях классов фильм и юзер.